plugins/debug/main.go

```
@@ -0,0 +1,147 @@
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type NetConf struct {
+	types.NetConf
+	CNIOutput  string     `json:"cniOutput,omitempty"`
+	AddHooks   [][]string `json:"addHooks,omitempty"`
+	DelHooks   [][]string `json:"delHooks,omitempty"`
+	CheckHooks [][]string `json:"checkHooks,omitempty"`
+}
+
+func main() {
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, bv.BuildString("none"))
+}
+
+func outputCmdArgs(fp io.Writer, args *skel.CmdArgs) {
+	fmt.Fprintf(fp, `ContainerID: %s
+Netns: %s
+IfName: %s
+Args: %s
+Path: %s
+StdinData: %s
+----------------------
+`,
+		args.ContainerID,
+		args.Netns,
+		args.IfName,
+		args.Args,
+		args.Path,
+		string(args.StdinData))
+}
+
+func parseConf(data []byte) (*NetConf, error) {
+	conf := &NetConf{}
+	if err := json.Unmarshal(data, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse")
+	}
+
+	return conf, nil
+}
+
+func getResult(netConf *NetConf) *type100.Result {
+	if netConf.RawPrevResult == nil {
+		return &type100.Result{}
+	}
+
+	version.ParsePrevResult(&netConf.NetConf)
+	result, _ := type100.NewResultFromResult(netConf.PrevResult)
+	return result
+}
+
+func executeHooks(netnsName string, hooks [][]string) {
+	netns, err := ns.GetNS(netnsName)
+	if err != nil {
+		return
+	}
+	defer netns.Close()
+
+	netns.Do(func(_ ns.NetNS) error {
+		for _, hookStrs := range hooks {
+			hookCmd := hookStrs[0]
+			hookArgs := hookStrs[1:]
+			//err := exec.Command(hookCmd, hookArgs...).Run()
+			output, err := exec.Command(hookCmd, hookArgs...).Output()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "OUTPUT: %v", output)
+				fmt.Fprintf(os.Stderr, "ERR: %v", err)
+			}
+		}
+		return nil
+	})
+}
+
+func cmdAdd(args *skel.CmdArgs) error {
+	netConf, _ := parseConf(args.StdinData)
+	// Output CNI
+	if netConf.CNIOutput != "" {
+		fp, _ := os.OpenFile(netConf.CNIOutput, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
```

is it safe to swallow error here?

---

I suppose so because plugins can proceed without output message.

---

this means `fp` could be nil at some times, and `fp.Close()` will bring a panic

---

gotcha. thank you for your comment! changed.